http://sourceforge.net/tracker/index.php?func=detail&aid=2902010&group_id=10894&atid=110894

Pulled from OE by: Saul Wold <saul.wold@intel.com>

--- ../generic/tclStrToD.c.orig
+++ ../generic/tclStrToD.c
@@ -71,9 +71,10 @@
 
 /*
  * MIPS floating-point units need special settings in control registers
- * to use gradual underflow as we expect.
+ * to use gradual underflow as we expect.  This fix is for the MIPSpro
+ * compiler.  
  */
-#if defined(__mips)
+#if defined(__sgi) && defined(_COMPILER_VERSION)
 #include <sys/fpu.h>
 #endif
 /*
@@ -2166,7 +2167,7 @@
     } bitwhack;
 #endif
 
-#if defined(__mips)
+#if defined(__sgi) && defined(_COMPILER_VERSION)
     union fpc_csr mipsCR;
 
     mipsCR.fc_word = get_fpc_csr();
